identity/handler.go

```diff
@@ -435,10 +436,13 @@ type AdminUpdateIdentityBody struct {
 //       500: jsonError
 func (h *Handler) update(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	var ur AdminUpdateIdentityBody
-	if err := errors.WithStack(jsonx.NewStrictDecoder(r.Body).Decode(&ur)); err != nil {
-		h.r.Writer().WriteError(w, r, err)
-		return
-	}
+        if err := errors.WithStack(jsonx.NewStrictDecoder(r.Body).Decode(&ur)); errors.Is(err, io.EOF) {
+                h.r.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Empty input json or file does not exist`).WithWrap(err)))
+                return
+        } else if err != nil {
+                h.r.Writer().WriteError(w, r, err)
+                return
+        }
```

Comment:
We could maybe update this and use `decoderx`.

Here is an example
https://github.com/ory/kratos/blob/686c9ba08ff1db8a310eaed5c4b3aec69e0f84da/selfservice/flow/logout/handler.go#L178-L183

Reply:
the `decoderx.Decode` function returns a herodot error for us and should also give an error for an empty body.

Reply:
Hi @Benehiko ,

Thanks for reviewing the code. I will try to incorporate your suggestions and push the changes soon.